### PR TITLE
cpu/cc2538: add periph_rtt

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds `periph_rtt` for cc2538. There is no overflow callback on cc2538, for now it is just set using the same compare ISR as for the alarm. This is handled similarly than `fe310` `CPU`. I think this should be pointed out in the API, I will open a subsequent PR addressing this.

### Testing procedure

```
BOARD=openmote-b make -C tests/periph_rtt/ flash test
```

```
/home/francisco/workspace/RIOT/dist/tools/pyterm/pyterm -p "/dev/riot/tty-openmote-b" -b "115200" --noprefix --no-repeat-command-on-empty-line
Connect to serial port /dev/riot/tty-openmote-b
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START

RIOT RTT low-level driver test
This test will display 'Hello' every 5 seconds

Initializing the RTT driver
RTT now: 102642
Setting initial alarm to now + 5 s (266407)
Done setting up the RTT, wait for many Hellos
Hello
Hello
Hello
Hello
Hello

make: Leaving directory '/home/francisco/workspace/RIOT/tests/periph_rtt'

```

### CODE REVIEW

File: cpu/cc2538/periph/rtt.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+void isr_sleepmode(void)
+{
+    if (alarm_cb) {
+        /* 'consume' the callback (as it might be set again in the cb) */
+        rtt_cb_t tmp = alarm_cb;
+        alarm_cb = NULL;
+        tmp(alarm_arg);
+    }
+    else if(overflow_cb) {
```
Comment (aabadie) [inline cpu/cc2538/periph/rtt.c]:
```suggestion
    else if (overflow_cb) {
```

File: cpu/cc2538/periph/rtt.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    alarm_arg = NULL;
+    irq_restore(irq);
+}
+
+void rtt_set_overflow_cb(rtt_cb_t cb, void *arg)
+{
+    unsigned irq = irq_disable();
+
+    /* set threshold to RTT_MAX_VALUE */
+    while(!(SMWDTHROSC_STLOAD & 0x1UL)) { }
```
Comment (aabadie) [inline cpu/cc2538/periph/rtt.c]:
```suggestion
    while (!(SMWDTHROSC_STLOAD & 0x1UL)) {}
```

File: cpu/cc2538/periph/rtt.c [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+void rtt_set_alarm(uint32_t alarm, rtt_cb_t cb, void *arg)
+{
+    assert(cb && !(alarm & ~RTT_MAX_VALUE));
+
+    unsigned irq = irq_disable();
+
+    /* set alarm value */
+    while(!(SMWDTHROSC_STLOAD & SMWDTHROSC_STLOAD_STLOAD_MASK)) { }
```
Comment (aabadie) [inline cpu/cc2538/periph/rtt.c]:
```suggestion
    while (!(SMWDTHROSC_STLOAD & SMWDTHROSC_STLOAD_STLOAD_MASK)) {}
```

File: cpu/cc2538/periph/rtt.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+{
+    rtt_poweron();
+}
+
+uint32_t rtt_get_counter(void)
+{
+    return ((SMWDTHROSC_ST0 & 0xFF) |
+           ((SMWDTHROSC_ST1 & 0xFF) << 8) |
+           ((SMWDTHROSC_ST2 & 0xFF) << 16) |
+           ((SMWDTHROSC_ST3 & 0xFF) << 24) );
```
Comment (aabadie) [inline cpu/cc2538/periph/rtt.c]:
```suggestion
           ((SMWDTHROSC_ST3 & 0xFF) << 24));
```

File: boards/openmote-b/Makefile.features [hunk 5]
```diff
@@ -7,3 +7,4 @@ FEATURES_PROVIDED += periph_i2c
 FEATURES_PROVIDED += periph_spi
 FEATURES_PROVIDED += periph_timer
 FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_rtt
```
Comment (aabadie) [inline boards/openmote-b/Makefile.features]: Please keep alphabetical order

File: cpu/cc2538/periph/rtt.c (hunk 2 above)
Comment (fjmolinas) [reply cpu/cc2538/periph/rtt.c]: Note to seel: use the macro

**Comment by fjmolinas** [comment]
Hmm yess I can prbably add a common rtt config file.

File: boards/openmote-b/include/periph_conf.h [hunk 6]
```diff
@@ -148,6 +148,24 @@ static const spi_conf_t spi_config[] = {
...
+#define RTT_MAX_VALUE       (0xffffffff)
+#if SYS_CTRL_OSC32K_USE_XTAL
+/* Frequency of XOSC off by default */
+#define RTT_FREQUENCY       (32768U)    /* in Hz. For changes see `rtt.c` */
+#else
+/* Frequency of RCOSC on by default */
+#define RTT_FREQUENCY       (32753U)    /* in Hz. For changes see `rtt.c` */
+#endif
+/** @} */
+
```
Comment (benpicco) [inline boards/openmote-b/include/periph_conf.h]: Does it make sense to make this board-specific?
It could as well go to `cpu/cc2538/include/periph_cpu.h`.

Comment (fjmolinas) [reply boards/openmote-b/include/periph_conf.h]: Makes sense!

**Comment by fjmolinas** [comment]
Rebased and pushed the fixups.

File: boards/openmote-b/Makefile.features [hunk 7]
```diff
@@ -4,6 +4,7 @@ CPU_MODEL = cc2538sf53
 # Put defined MCU peripherals here (in alphabetical order)
 FEATURES_PROVIDED += periph_adc
 FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_rtt
```
Comment (aabadie) [inline boards/openmote-b/Makefile.features]: Maybe also move this in cpu/cc2538/Makefile.features ?

**Comment by aabadie** [comment]
@fjmolinas, Travis complains because of a missing doxygen group. Can you fix this ? You can squash directly.

**Pull request merged**